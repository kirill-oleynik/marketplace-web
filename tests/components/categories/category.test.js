const React = require('react');
const { shallow } = require('enzyme');
const toJSON = require('enzyme-to-json').default;
const {
  Category
} = require('../../../app/components/categories/category');

describe('#render', () => {
  describe('when it has less than preview amount apps', () => {
    test('it renders correctly', () => {
      const component = shallow(
        <Category
          t={(translation) => translation}
          expand={() => {}}
          collapse={() => {}}
          category={{
            id: 1,
            title: 'Test1',
            slug: 'test_1',
            applications: [],
            isFetched: false,
            isFetchable: false
          }}
        />
      );

      expect(
        toJSON(component)
      ).toMatchSnapshot();
    });
  });

  describe('when it has more than preview amount apps', () => {
    describe('and there are apps to fetch', () => {
      test('it renders correctly', () => {
        const component = shallow(
          <Category
            t={(translation) => translation}
            expand={() => {}}
            collapse={() => {}}
            category={{
              id: 1,
              title: 'Test1',
              slug: 'test_1',
              applications: [],
              isFetched: false,
              isFetchable: true
            }}
          />
        );

        expect(
          toJSON(component)
        ).toMatchSnapshot();
      });
    });

    describe('and there is no apps to fetch', () => {
      test('it renders correctly', () => {
        const component = shallow(
          <Category
            t={(translation) => translation}
            expand={() => {}}
            collapse={() => {}}
            category={{
              id: 1,
              title: 'Test1',
              slug: 'test_1',
              applications: [],
              isFetched: true,
              isFetchable: true
            }}
          />
        );

        expect(
          toJSON(component)
        ).toMatchSnapshot();
      });
    });
  });
});
