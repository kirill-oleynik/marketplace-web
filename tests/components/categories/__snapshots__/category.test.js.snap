// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render when it has less than preview amount apps it renders correctly 1`] = `
<div
  className="mb-20"
  id="test_1"
>
  <div
    className="mb-30"
  >
    <h3
      className="font-20 font-700"
    >
      Test1
    </h3>
  </div>
  <Row
    tag="div"
  />
  <div
    className="text-center text-md-left"
  />
</div>
`;

exports[`#render when it has more than preview amount apps and there are apps to fetch it renders correctly 1`] = `
<div
  className="mb-20"
  id="test_1"
>
  <div
    className="mb-30"
  >
    <h3
      className="font-20 font-700"
    >
      Test1
    </h3>
  </div>
  <Row
    tag="div"
  />
  <div
    className="text-center text-md-left"
  >
    <ButtonWithIcon
      className="mb-10"
      icon="arrow-down"
      onClick={[Function]}
      text="categories.showMore"
      type="button"
    />
  </div>
</div>
`;

exports[`#render when it has more than preview amount apps and there is no apps to fetch it renders correctly 1`] = `
<div
  className="mb-20"
  id="test_1"
>
  <div
    className="mb-30"
  >
    <h3
      className="font-20 font-700"
    >
      Test1
    </h3>
  </div>
  <Row
    tag="div"
  />
  <div
    className="text-center text-md-left"
  >
    <ButtonWithIcon
      className="mb-10"
      icon="arrow-up"
      onClick={[Function]}
      text="categories.showLess"
      type="button"
    />
  </div>
</div>
`;
